Library/Homebrew/download_strategy.rb

```diff
@@ -532,6 +532,82 @@ def _fetch
   end
 end
 
+# GitHubReleaseDownloadStrategy downloads tarballs from GitHub Release assets.
+# To use it, add ":using => GitHubReleaseDownloadStrategy" to the URL section
+# of your formula. This download strategy uses GitHub access tokens (in the
+# environment variables GITHUB_TOKEN) to sign the request.
+# This strategy is suitable for corporate use just like S3DownloadStrategy,
+# because it lets you use a private GttHub repository for internal distribution.
+# It works with public one, but in that case simply use CurlDownloadStrategy.
+class GitHubReleaseDownloadStrategy < CurlDownloadStrategy
+  require 'open-uri'
+
+  def initialize(name, resource)
+    super
+
+    @github_token = ENV["GITHUB_TOKEN"]
+    unless @github_token
+      puts "Environmental variable GITHUB_TOKEN is required."
+      raise CurlDownloadStrategyError, @url
+    end
+
+    url_pattern = %r|https://github.com/(\S+)/(\S+)/releases/download/(\S+)/(\S+)|
```

Comment:
@MikeMcQuaid I am trying to support this, but the following URL does not contain any asset.
https://api.github.com/repos/Homebrew/brew/releases/tags/1.1.5

How do we download it? I could not find an GitHub API to get archive. Are there any good ideas?